### PR TITLE
Adds onlyEssentialData param to case queries

### PR DESCRIPTION
**Pull request by murilovmachado**
## Description
Adds `onlyEssentialData` optional param to:
- `GET /cases/` (Case List)
- `POST /cases/search` (Search)
- `GET /cases/:id` (Fetch single case)

This is implemented by 2 different pieces:
- Changes to the queries themselves, so they avoid unnecessary joins
- Mapping to the queries' result, to omit even more fields
  - This could be done within the queries, but it would cause more harm than good

### Related Issues
Fixes https://tech-matters.atlassian.net/browse/CHI-2423

### Verification steps
Open the case list and verify that the network call brings only essential data.

### CODE REVIEW

File: hrm-domain/hrm-core/case/sql/case-get-sql.ts [hunk 1]
```diff
@@ -25,26 +25,47 @@ export const selectContactsOwnedCount = (ownerVariableName: string) =>
    FROM "Contacts" 
    WHERE "caseId" = cases.id AND "accountSid" = cases."accountSid" AND "twilioWorkerId" = $<${ownerVariableName}>`;
 
-export const selectSingleCaseByIdSql = (tableName: string) => `SELECT
+const leftJoinLateralContacts = (onlyEssentialData?: boolean) => {
+  if (onlyEssentialData) {
+    return `
+      LEFT JOIN LATERAL (
+        SELECT COALESCE(jsonb_agg(to_jsonb(c)), '[]') AS  "connectedContacts"
```
Comment (stephenhand) [inline hrm-domain/hrm-core/case/sql/case-get-sql.ts:32]: We could potentially request less from the contact table here, but this is fine for now, avoiding the extra joins is the real win

File: hrm-domain/hrm-core/unit-tests/case/caseService.test.ts [hunk 2]
```diff
@@ -272,6 +273,7 @@ describe('searchCases', () => {
           user: twilioUser(workerSid, []),
           permissions: rulesMap.open,
         },
```
Comment (stephenhand) [inline hrm-domain/hrm-core/unit-tests/case/caseService.test.ts:275]: An extra test or two verifying the behaviour for both flag settings might be nice, not blocking tho

**Review by stephenhand — APPROVED** [review]

**Comment by stephenhand** [comment]
Looks great!

**Comment by stephenhand** [comment]
@murilovmachado - is this PR everything for this ticket or do we need the timeline / case section list endpoints as well?

**Comment by murilovmachado** [comment]
> @murilovmachado - is this PR everything for this ticket or do we need the timeline / case section list endpoints as well?

@stephenhand This PR covers everything we need for now.
- `GET /cases/:id` still gets all sections and all the data for activities.
  - it just don't bring additional connected contact info such as referrals, csam, conversation media
  - This is the endpoint we'll call when navigating to the Case page
  - We plan on having the activities being loaded from a separate endpoint, but this is out-of-scope here and not needed for now
- `GET /cases/` and `POST /cases/search` are the ones that bring way less data
  - They are called on Case List and Search Cases

File: hrm-domain/hrm-core/unit-tests/case/caseService.test.ts (hunk 2 above)
Comment (murilovmachado) [reply hrm-domain/hrm-core/unit-tests/case/caseService.test.ts:337]: I've actually got an edge case bug while writing the unit tests xD
